File: Bit manipulation/136. Single Number.java
class Solution {
    /*
    XOR used here....if a=b then a^b=1 so
    if nums= 4,1,2,1,2
    ans=4^1^2^1^2
    here 2^2 and 1^1 both will be cancel out and 4 is only one so it will not be cancelled out
    */
    public int singleNumber(int[] nums) {
        int result = 0;
        for(int num: nums){
            result = result ^ num;
        }
        return result;
    }
}

File: Bit manipulation/136. Single Number.cpp
class Solution
{
public:
    /*
    n ^ n -> 0
    n ^ 0 -> n
    -> remainder

    */
    int singleNumber(vector<int> &nums)
    {
        int res = 0;
        for (int n : nums)
        {
            res = res ^ n;
        }
        return res;
    }
};